airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/TruncateInsertIntoConsumer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class TruncateInsertIntoConsumer implements TmpToFinalTable {
+
+  private final InsertTableOperations destination;
+  private Map<String, DestinationCopyContext> copyConfigs;
+
+  public TruncateInsertIntoConsumer(InsertTableOperations destination) {
+    this.destination = destination;
+    this.copyConfigs = new HashMap<>();
+  }
+
+  @Override
+  public void setContext(Map<String, DestinationCopyContext> configs) {
+    copyConfigs = configs;
+  }
+
+  @Override
+  public void execute() throws Exception {
+    for (Entry<String, DestinationCopyContext> entry : copyConfigs.entrySet()) {
```

Comment:
given the lifecycle, you should have a Precondition in the execute so that it is "documented" copyConfigs should be set through setContext